core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -47,11 +46,13 @@ class Partition(val topic: String,
                 val partitionId: Int,
                 time: Time,
                 replicaManager: ReplicaManager) extends Logging with KafkaMetricsGroup {
+
   val topicPartition = new TopicPartition(topic, partitionId)
 
-  private val localBrokerId = replicaManager.config.brokerId
-  private val logManager = replicaManager.logManager
-  private val zkUtils = replicaManager.zkUtils
+  // replicaManager will be null only if this partition is ReplicaManager.OfflinePartition
```

Comment:
Initially I wanted to avoid adding complexity to the constructor. Sure, I have added isOffline flag to the constructor.